### PR TITLE
Add asserts and additional tests

### LINKED ISSUE #131: Potential assertion error
These functions are missing `@assertion` macros to avoid matrices` and `c` to have different lenghts from `y`

```julia
    function StateSpaceModel(y::VecOrMat{Typ}, Z::Array{Typ, 3}, T::Matrix{Typ}, R::Matrix{Typ}, 
                             d::Matrix{Typ}, c::Matrix{Typ}, H::Matrix{Typ}, Q::Matrix{Typ}) where Typ <: Real

        # Convert y to a Matrix
        y = ensure_is_matrix(y)
        # Build StateSpaceDimensions
        dim = build_ss_dim(y, Z, T, R)
        return new{Typ}(y, Z, T, R, d, c, H, Q, dim, find_missing_observations(y), "time-variant")
    end

    function StateSpaceModel(y::VecOrMat{Typ}, Z::Matrix{Typ}, T::Matrix{Typ}, R::Matrix{Typ}, 
                             d::Matrix{Typ}, c::Matrix{Typ}, H::Matrix{Typ}, Q::Matrix{Typ}) where Typ <: Real

        # Convert y to a Matrix
        y = ensure_is_matrix(y)
        # Build StateSpaceDimensions
        dim = build_ss_dim(y, Z, T, R)
        Zvar = Array{Typ, 3}(undef, dim.p, dim.m, dim.n)
        for t in 1:dim.n, i in axes(Z, 1), j in axes(Z, 2)
            Zvar[i, j, t] = Z[i, j]
        end

        return new{Typ}(y, Zvar, T, R, d, c, H, Q, dim, find_missing_observations(y), "time-invariant")
    end
```

### PR DESCRIPTION
**Pull request by raphaelsaavedra**
Closes #131

### CODE REVIEW

File: src/structures.jl [hunk 1]
```diff
@@ -113,34 +113,46 @@ struct StateSpaceModel{Typ <: Real}
...
                              d::Matrix{Typ}, c::Matrix{Typ}, H::Matrix{Typ}, Q::Matrix{Typ}) where Typ <: Real
 
         # Convert y to a Matrix
         y = ensure_is_matrix(y)
+
         # Build StateSpaceDimensions
         dim = build_ss_dim(y, Z, T, R)
+
+        # Ensure dimensions are coherent
+        assert_dimensions(c, d, dim)
```
Comment (guilhermebodin) [inline src/structures.jl]: here it is a little tricky, if Z has more entries than y, c and d should also have more entries.

Comment (raphaelsaavedra) [reply src/structures.jl]: Yep that's why the `assert_dimensions` checks if the size is `>= n` rather than `==`

Comment (guilhermebodin) [reply src/structures.jl]: yes but what if Z, c and d have different `n`s ?

Comment (raphaelsaavedra) [reply src/structures.jl]: Yeah this only prevents problems in the estimation, not in the forecasting. I didn't worry too much about it but I can change it to check if Z is larger than n and if it is verify if c and d have the same size as Z

**Pull request merged**